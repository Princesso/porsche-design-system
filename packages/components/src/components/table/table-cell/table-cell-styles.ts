import { getCss } from '../../../utils';
import { addImportantToEachRule, getThemedColors, hostHiddenStyles } from '../../../styles';
import { spacingFluidSmall } from '@porsche-design-system/utilities-v2';

export const getComponentCss = (multiline: boolean): string => {
  return getCss({
    '@global': {
      ':host': {
        ...addImportantToEachRule({
          display: 'table-cell',
          padding: spacingFluidSmall,
          margin: 0,
          borderBottom: `1px solid ${getThemedColors('light').contrastLowColor}`,
          whiteSpace: multiline ? 'normal' : 'nowrap',
          ...hostHiddenStyles,
          transform: 'translate3d(0,0,0)', // Change stacking context for hover state
        }),
        verticalAlign: 'middle',
      },
    },
  });
};
