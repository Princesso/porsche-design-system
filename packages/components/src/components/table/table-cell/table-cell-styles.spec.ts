import { getComponentCss } from './table-cell-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    [true, 'dark'],
    [false, 'light'],
  ])('should return correct css for multiline: %s and theme: %s', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
