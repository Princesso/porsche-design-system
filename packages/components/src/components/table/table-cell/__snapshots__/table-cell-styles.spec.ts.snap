// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for multiline: false 1`] = `
":host {
  display: table-cell !important;
  padding: 0.75rem !important;
  margin: 0 !important;
  border-bottom: 1px solid #e3e4e5 !important;
  white-space: nowrap !important;
  vertical-align: middle;
}"
`;

exports[`getComponentCss() should return correct css for multiline: true 1`] = `
":host {
  display: table-cell !important;
  padding: 0.75rem !important;
  margin: 0 !important;
  border-bottom: 1px solid #e3e4e5 !important;
  white-space: normal !important;
  vertical-align: middle;
}"
`;
